demos/yielding/README.md

```diff
@@ -0,0 +1,47 @@
+# Yielding
+
+In this demo we show the usage of yielding, which occurs in a computer program during multithreading.
+
+By calling syscall `yield`, the thread gives away the utilization of the CPU to another thread, or a specific thread (discussed in another demo).
+
+To describe thread yielding, we need to introduce:
+
+* [`add_thread_from_worker`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L42)
+* [`do_scheduler`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/kernel/thread.c#L353)
+* [`yield`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L92)
+
+## [`add_thread_from_worker`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L42)
+
+In AMT we store all of the threads in an array called `all_threads`.
+
+When a thread calls this function, the kernel will create a new thread with the given arguments, initialize the permissions, and set its starting point to be the given function (would be `thread_worker()` in this demo).
+
+After the creation of the thread, it will wait to be chosen by the scheduler so it can utilize the CPU.
+
+## [`do_scheduler`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/kernel/thread.c#L353)
+
+The job of the scheduler is to choose the next thread to make use of the CPU. 
+
+In AMT, the scheduler simply walks through the `all_threads` array to find the next thread.
+
+Then the task of the thread to be switched will be suspended until the thread has been chosen to run.
+
+## [`yield`](https://github.com/DavidSpickett/ARMMultiTasking/blob/master/src/user/thread.c#L92)
+
+To yield the utilization, the thread calls the syscall `yield` and sets the kind of yielding as `YIELD_ANY`. The scheduler will choose the next thread to run itself.
+
+The current thread will have to wait until it is yielded to, to run again.
+
+## The progress of yielding demo
```

Comment:
I think there's a better way to title this section.
"Walkthrough" perhaps? We can leave out the demo's name, that's obvious from context.